src/kinetics/KineticsFactory.cpp

```diff
@@ -168,17 +171,31 @@ void addReactions(Kinetics& kin, const AnyMap& phaseNode, const AnyMap& rootNode
             AnyMap reactions = AnyMap::fromYamlFile(fileName,
                 rootNode.getString("__file__", ""));
             for (const auto& R : reactions[node].asVector<AnyMap>()) {
-                kin.addReaction(newReaction(R, kin));
+                try {
+                    kin.addReaction(newReaction(R, kin));
+                } catch (CanteraError& err) {
+                    std::string captured_err = err.what();
+                    format_to(add_rxn_err, captured_err);
```

Comment:
* You can just write `err.what()` directly in the call to `format_to`. There's no need to create a variable for a string that's only used once.
* This is kind of a subtle point, but the signature of `format_to` is actually `format_to(buffer, format string, arguments...)`. What you've written is only valid in the special case where `captured_err` doesn't contain anything that might be interpreted as a format group, i.e. something containing `{` and `}`. For some validation errors, this definitely happens, which results in an error here. The correct syntax for this is actually `format_to(add_rxn_err, "{}", captured_err)`.

To catch this mistake, I would suggest adding a test for a different kind of reaction error, such as an unbalanced reaction equation.